src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -97,38 +103,49 @@ public void setup() {
         customShelfService.save(customShelf);
 
         Assumptions.assumeTrue(predefinedShelfService != null);
-        bookForm = createBookForm(true);
-
+        bookForm = createBookForm(READ, true);
     }
 
-    private BookForm createBookForm(boolean isInSeries) {
+    private BookForm createBookForm(ShelfName shelf, boolean isInSeries) {
         BookForm bookForm = new BookForm(predefinedShelfService, customShelfService);
-        bookForm.setBook(createBook(predefinedShelfService, isInSeries));
+        readShelf = predefinedShelfService.findAll().get(2);
+        bookForm.setBook(createBook(shelf, isInSeries));
         return bookForm;
-
     }
 
-    private Book createBook(PredefinedShelfService predefinedShelfService, boolean isInSeries) {
+    private Book createBook(ShelfName shelfName, boolean isInSeries) {
+        // values that are present in all forms
```

Comment:
Could you please remove the comment? I understand why you've put in, but I don't think it adds much value. A reader can see that a `Book` can be constructed with only a `title`, `author` and `PredefinedShelf` by virtue of our public constructor.

Reply:
done